### DISCUSSION TITLE
What is the recommended/ideal way to test component interactions?

### DISCUSSION THREAD

**Discussion by Axedyson** [storybookjs/storybook#16726] [Help] [answered]
I see that there are now two different ways to to test component interactions, the first is using jest and testing library [manually so to speak](https://medium.com/storybookjs/how-to-test-component-interactions-dd0ee76a7ee7).
The second and new option is using a feature called [addon-interactions](https://github.com/storybookjs/storybook/tree/next/addons/interactions) which is also described here: https://medium.com/storybookjs/interactive-stories-beta-14572676db9a.
My question is which method should I choose? And how would I run the addon-interactions (https://github.com/storybookjs/storybook/tree/next/addons/interactions) in CI/CD using github actions/github workflows to prevent regressions? Also how would I use addon-interactions along side Accessibility testing using jest-axe or maybe there is a better way? Thanks!

**Comment by shilman** [comment, marked as answer]
We're still building it!

6.4's interactive stories are the foundation for this. In 6.5, we will release a Jest-based test runner that makes them easy to run these tests in CI. We plan to extend the test runner with other kinds of tests in subsequent releases. We hope this will become a simple yet powerful way to test UI in isolation.

Where does this leave `@storybook/testing-react/vue/etc`? We will continue to maintain improve this as well. Users who don't mind a more manual approach will have more flexibility to work with other test runners, different test styles, and so on. Storybook strives for broad compatibility, and this is how we plan to achieve that in testing.

What should you do today? If you need something immediately, the "manual" approach is your only option. The 6.4 release (currently in RC) is intended to give our community a taste of what's to come and help us to collect feedback on this direction. If you want to help shape this direction and don't mind using experimental software I'd definitely recommend kicking the tires.

**Reply by slugmandrew** [reply]
This is great to hear! Am I right in thinking this would let me call the play function manually and include it in my storyshots snap? If I have understood correctly then that's exactly what I need.

Looking forward to 6,5!
